CIME/XML/namelist_definition.py

```diff
@@ -215,7 +219,7 @@ def get_value_match(self, vid, attributes=None, exact_match=True, entry_node=Non
         # NOTE(wjs, 2021-06-04) In the following call, replacement_for_none='' may not
         # actually be needed, but I'm setting it to maintain some old logic, to be safe.
         value = super(NamelistDefinition, self).get_value_match(
-            vid.lower(),
+            vid,
```

Comment:
The removal of `.lower()` here and in the call from `get_default_value` aren't needed to get things to work, but I think could be needed in principle in the future: I think these weren't needed because (1) `get_default_value` doesn't seem to ever be called, and (2) it happens that the `vid` here is irrelevant because we always pass an `entry_node` into this call.

I tested removing the setting of `entry_node` a couple of lines above. In this case, I got an error in `get_value_match` unless I removed the `lower` here, when working with mixed-case variable names.